Automatically turn off Slf4jLogsafeArgs in test code

## What happened

I just got an error message from `Slf4jLogsafeArgs` in conjure-java-runtime tests:

```
> Task :okhttp-clients:compileTestJava
/Users/dfox/conjure-java-runtime/okhttp-clients/src/test/java/com/palantir/conjure/java/okhttp/FlowControlTest.java:89: warning: [Slf4jLogsafeArgs] slf4j log statement does not use logsafe parameters for arguments [1, 2]
            log.info("Average rate is {}, 1 minute rate is {}", rate.getMeanRate(), rate.getOneMinuteRate());
                    ^
    (see https://github.com/palantir/gradle-baseline#baseline-error-prone-checks)
```

## What should have happened

This should never have wasted my time because it's test-only code so I don't care if it's a logsafe arg.

--- a/baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/Slf4jLogsafeArgs.java
+++ b/baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/Slf4jLogsafeArgs.java
@@ -73,7 +73,7 @@ public final class Slf4jLogsafeArgs extends BugChecker implements MethodInvocati
         }
         List<Integer> badArgs = badArgsBuilder.build();
 
-        if (badArgs.isEmpty()) {
+        if (badArgs.isEmpty() || TestCheckUtils.isTestCode(state)) {
             return Description.NO_MATCH;
         } else {
             return buildDescription(tree)

--- a/changelog/@unreleased/pr-756.v2.yml
+++ b/changelog/@unreleased/pr-756.v2.yml
@@ -1,0 +1,5 @@
+type: improvement
+improvement:
+  description: Disable errorprone `Slf4jLogsafeArgs` in test code
+  links:
+  - https://github.com/palantir/gradle-baseline/pull/756

